tests/api/options.php

```diff
@@ -0,0 +1,78 @@
+<?php
+/**
+ * Options REST API Test
+ *
+ * @package WooCommerce Admin\Tests\API
+ */
+
+use \Automattic\WooCommerce\Admin\API\Options;
+
+/**
+ * WC Tests API Options
+ */
+class WC_Tests_API_Options extends WC_REST_Unit_Test_Case {
+
+	/**
+	 * Endpoints.
+	 *
+	 * @var string
+	 */
+	protected $endpoint = '/wc-admin/v1/options';
+
+	/**
+	 * Setup test data. Called before every test.
+	 */
+	public function setUp() {
+		parent::setUp();
+
+		$this->user = $this->factory->user->create(
+			array(
+				'role' => 'administrator',
+			)
+		);
+	}
+
+	/**
+	 * Test getting options without valid permissions.
+	 *
+	 * @since 3.5.0
+	 */
+	public function test_get_options_without_permission() {
+		wp_set_current_user( 0 );
+		$request = new WP_REST_Request( 'GET', $this->endpoint );
+		$request->set_query_params( array( 'options' => 'woocommerce_demo_store_notice' ) );
+		$response = $this->server->dispatch( $request );
+		$this->assertEquals( 401, $response->get_status() );
+	}
+
+	/**
+	 * Test that options can be retrieved.
+	 */
+	public function test_get_options() {
+		wp_set_current_user( $this->user );
+
+		$request = new WP_REST_Request( 'GET', $this->endpoint );
+		$request->set_query_params( array( 'options' => 'woocommerce_demo_store_notice' ) );
+		$response = $this->server->dispatch( $request );
+		$data     = $response->get_data();
+
+		$this->assertEquals( 200, $response->get_status() );
+		$this->assertEquals( get_option( 'woocommerce_demo_store_notice' ), $data['woocommerce_demo_store_notice'] );
+	}
+
+	/**
+	 * Test that options can be updated.
+	 */
+	public function test_update_options() {
```

Comment:
Should we have a `test_update_options_without_permission` for completeness?

Reply:
Added in Updated in https://github.com/woocommerce/woocommerce-admin/pull/3004/commits/ba72b7eb457ba83a6b14966c768ba3d2ea8b8c99